index.js

```diff
@@ -369,158 +369,98 @@
                         self.sanctuaryTypeIdentifiers);
   }
 
-} (function($, Either, Maybe, Pair, show, Z, type) {
+}) (($, Either, Maybe, Pair, show, Z, type) => {
 
   'use strict';
 
-  /* istanbul ignore if */
-  if (typeof __doctest !== 'undefined') {
-    /* eslint-disable no-unused-vars */
-    var Descending = __doctest.require ('sanctuary-descending');
-    var Nil = (__doctest.require ('./test/internal/List')).Nil;
-    var Cons = (__doctest.require ('./test/internal/List')).Cons;
-    var Sum = __doctest.require ('./test/internal/Sum');
-    var S = (function(S) {
-      var S_ = S.create ({
-        checkTypes: true,
-        env: S.env.concat ([
-          (__doctest.require ('./test/internal/List')).Type ($.Unknown),
-          Sum.Type
-        ])
-      });
-      S_.env = S.env;  // see S.env doctest
-      return S_;
-    } (require ('.')));
-    /* eslint-enable no-unused-vars */
-  }
-
-  //  Left :: a -> Either a b
-  var Left = Either.Left;
-
-  //  Right :: b -> Either a b
-  var Right = Either.Right;
-
-  //  Nothing :: Maybe a
-  var Nothing = Maybe.Nothing;
-
-  //  Just :: a -> Maybe a
-  var Just = Maybe.Just;
-
-  //  B :: (b -> c) -> (a -> b) -> a -> c
-  function B(f) {
-    return function(g) {
-      return function(x) {
-        return f (g (x));
-      };
-    };
-  }
+  const {
+    Left,
+    Right,
+  } = Either;
 
-  //  C :: (a -> b -> c) -> b -> a -> c
-  function C(f) {
-    return function(y) {
-      return function(x) {
-        return f (x) (y);
-      };
-    };
-  }
+  const {
+    Nothing,
+    Just,
+  } = Maybe;
 
-  //  curry2 :: ((a, b) -> c) -> a -> b -> c
-  function curry2(f) {
-    return function(x) {
-      return function(y) {
-        return f (x, y);
-      };
-    };
-  }
+  const {
+    fst,
+    snd,
+    swap,
+  } = Pair;
 
-  //  curry3 :: ((a, b, c) -> d) -> a -> b -> c -> d
-  function curry3(f) {
-    return function(x) {
-      return function(y) {
-        return function(z) {
-          return f (x, y, z);
-        };
-      };
-    };
+  let Descending, Nil, Cons, Sum, S;
+  /* istanbul ignore if */
+  if (typeof __doctest !== 'undefined') {
+    const {create, env} = __doctest.require ('.');
+    const List = __doctest.require ('./test/internal/List');
+    Descending = __doctest.require ('sanctuary-descending');
+    ({Nil, Cons} = List);
+    Sum = __doctest.require ('./test/internal/Sum');
+    S = Object.assign (
+      create ({
+        checkTypes: true,
+        env: env.concat ([List.Type ($.Unknown), Sum.Type]),
+      }),
+      {env}  // see S.env doctest
+    );
+    [Descending, Nil, Cons, Sum, S];
```

Comment:
These identifiers would otherwise only be referenced in doctests. Referencing them here avoids lint errors.

Reply:
Maybe warrants a comment